src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/VillagerTradingListener.java

```diff
@@ -48,12 +48,12 @@ public void onPreTrade(InventoryClickEvent e) {
             }
 
             if (e.getResult() == Result.DENY) {
-                Slimefun.getLocalization().sendMessage((Player) e.getWhoClicked(), "villagers.no-trading", true);
+                Slimefun.getLocalization().sendMessage(e.getWhoClicked(), "villagers.no-trading", true);
```

Comment:
I don't think we can do this here due to backwards compatibility.

Reply:
I looked through the javadoc from 1.16 to 1.19, the `getWhoClicked()` always returns a `HumanEntity`, which is a subinterface of `CommandSender`, so casting here isn't needed.

Reply:
Do you remember why you add this casting?

Reply:
Will resolve this as stale, backwards compat is dead and 1.16 on you said you checked so it's fine by me